### PR TITLE
Add/chart dashboard block

### LINKED ISSUE #690: Component: Dashboard Charts
This description is lifted from the dashboard overview post at: p6riRB-3qG-p2

Primary interactions for the graphs (aggregation, graph type, and settings menu) have been moved into the block header, and their selections apply to all of the graphs that are active in the block. The block features a settings menu that allows users to show or hide individual graphs, change the color palette, or remove the block. 

screen shot 2018-08-09 at 4 25 09 pm

Interactive features on individual graphs such as hover and tap states are disabled. Users are also unable to turn a series off or on in the legend. [note the legend shows a colored box without a check] Instead, the entire graph component is clickable, and directs users to the associated report.

screen shot 2018-08-09 at 6 34 26 pm

Missing from the image above would be a focus state, which would wrap an entire graph component. 

**Data available to graph: (as the reports become available)** 

*bold denotes default selection

Gross Revenue > Revenue report – Gross revenue tab selected
**Net Revenue > Revenue Report – Net Revenue tab selected
Orders > Order Report – Orders tab selected**
Average Order Value > Order Report – AOV tab selected
Items Sold > Product Report – Items Sold tab selected
Refunds > Orders Report – Refunds Tab selected Filtered by refunded orders
Discounted Orders > Coupons Report – Discounted Orders tab selected
Gross Discounted > Coupons Report – Gross Discounted tab selected
Total Tax > Tax Report > Total Tax tab selected
Order Tax > Tax Report > Order Tax tab selected
Shipping Tax > Tax Report > Shipping Tax tab selected
Shipping > Revenue Report – Shipping tab active
Downloads > Download Report
Low Stock > Stock Report – Filtered by low stock products
Out of Stock > Stock Report – Filtered by out of stock products
New customers > Orders report – Filtered by new customers
Returning customers > Orders Report  – filtered by returning customers

### PR DESCRIPTION
**Pull request by greenafrican**
Fixes #690 

This PR introduces a `chart-block` to the dashboard, which wraps the `report-chart` component. The PR introduces a new `mode='block'` to the `report-chart` props to ensure that the legend is always fixed below the chart.

block

### Detailed test instructions:

- Open WooCommerce > Dashboard
- Ensure that the legend is on the bottom
- Ensure that there are no chart options in the chart header (line/bar type or interval selector)
- Hover over the chart to show the tooltip

### CODE REVIEW

File: client/dashboard/chart-block/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				label: 'Orders Count',
+				type: 'number',
+			},
+		];
+		const path = '/analytics/orders';
+		const query = {
+			period: 'quarter',
+			compare: 'previous_period',
+			type: 'line',
+		};
```
[the remark this answers is not in the document]
Comment (greenafrican) [reply client/dashboard/chart-block/index.js]: Good point. I'll move them into an example file.

**Comment by timmyc** [comment]
@jeffstieler could you give this another pass?

**Comment by jeffstieler** [comment]
@timmyc I'm not seeing any change in the area I commented on.

**Comment by timmyc** [comment]
Sorry about that, I thought some more commits had been made @jeffstieler

File: packages/components/src/chart/d3chart/legend.js [hunk 2]
```diff
@@ -158,6 +163,7 @@ D3Legend.propTypes = {
 };
 
 D3Legend.defaultProps = {
+	interactive: true,
```
Comment (timmyc) [inline packages/components/src/chart/d3chart/legend.js]: We will want to add some documentation to the CHANGELOG for the components module to reflect the updates.

**Review by timmyc — COMMENTED** [review]
Wow. This is awesome. Couple of minor comments, but seriously Rob thanks for this PR - this is bringing the dashboard to the next level.

image

File: client/dashboard/dashboard-charts/block.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import { Component, Fragment } from '@wordpress/element';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { Card } from '@woocommerce/components';
+import ReportChart from 'analytics/components/report-chart';
+import './block.scss';
+class ChartBlock extends Component {
```
Comment (justinshreve) [inline client/dashboard/dashboard-charts/block.js]: Should we make a dashboard block/chart component available under the components package or is that not useful yet? I can't recall how extendable we are making the dashboard for v1, but this seems like something reusable with the `endpoint` prop for example.

File: client/dashboard/dashboard-charts/config.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+
+import { charts as ordersCharts } from '../../analytics/report/orders/config';
+import { charts as productsCharts } from '../../analytics/report/products/config';
+import { charts as revenueCharts } from '../../analytics/report/revenue/config';
```
Comment (justinshreve) [inline client/dashboard/dashboard-charts/config.js]: I think for these imports you can just do `analytics/report/revenue/config`, etc.

File: client/dashboard/dashboard-charts/index.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+import { ToggleControl, IconButton, NavigableMenu } from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { EllipsisMenu, MenuItem, SectionHeader } from '@woocommerce/components';
```
Comment (justinshreve) [inline client/dashboard/dashboard-charts/index.js]: I believe we've been moving this to a new 

```
/**
 * WooCommerce dependencies
 */
```

section.

File: package-lock.json [hunk 6]
```diff
@@ -3192,7 +3192,7 @@
     },
     "ansi-escapes": {
       "version": "3.1.0",
-      "resolved": "https://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
+      "resolved": "http://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
```
Comment (justinshreve) [inline package-lock.json]: I think the `package-lock.json` changes can be omitted and will make this PR a bit smaller.

**Comment by timmyc** [comment]
@justinshreve and @jeffstieler I applied some changes to this PR based upon the feedback. The one main addition was creating a set of "default" charts to show for now:

image

I added a TODO for making chart selections persist akin to how table columns are now persisted.

I also have a few follow-up issues to make after this PR:

- Add hover state for entire chart block
- Add click-action for entire chart block to load the selected chart/summary number combo
- Bug in y-axis when only quantity one is needed and axis displays tick marks for `1, 2, 2` and plots quantity 1 at the first 2 tick mark.
- Can't read properly `current` of undefined in legend.js line 38

**Review by timmyc — APPROVED** [review]
I'm going to land this PR since it is pretty awesome and would like it to be in the next release. Going to log a follow-up issue to look at tidying up how things look in a narrow viewport:

image

**Pull request merged**